### PR TITLE
fix: parse scientific notation numbers in expression

### PR DESCRIPTION
**Pull request by markusrf**
The expression lexer was missing parsing for scientific notation (e.g. `1.23e-05` or `3.4e7`) for numbers part of an expressions with multiple tokens.

Ex. expression `2 {*} 3.4e-05` would be parsed as tokens `[2, {*}, 3.4, e-05]`, which would give issues as there are no variables named e-05.

### CODE REVIEW

File: src/ecalc/libraries/libecalc/common/libecalc/expression/expression_evaluator.py [hunk 1]
```diff
@@ -325,6 +325,7 @@ def lexer(expression: Union[str, int, float]) -> List[Token]:
         (r"while", TokenTag.operator),
         (r"do", TokenTag.operator),
         (r"end", TokenTag.operator),
+        (r"[0-9]\.[0-9]+e-?[0-9]+", TokenTag.numeric),  # Scientific notation, e.g. 1.23e-05 or 3.4e7
```
Comment (TeeeJay) [inline src/ecalc/libraries/libecalc/common/libecalc/expression/expression_evaluator.py:328]: what about 1e-8? It would. Can this happen to big numbers as well? In that case 3e29 and 3e+29?

File: src/ecalc/libraries/libecalc/common/libecalc/expression/expression_evaluator.py [hunk 2]
```diff
@@ -356,7 +357,7 @@ def check_tokens(tokens):
     if str(first_token) in list(OPERATORS.keys()):
         raise ValueError(f"Expression ({var}) can not start with an operator")
     if str(last_token) in list(OPERATORS.keys()):
-        raise ValueError(f"Expression ({var}) can not start with an operator")
+        raise ValueError(f"Expression ({var}) can not end with an operator")
```
Comment (TeeeJay) [inline src/ecalc/libraries/libecalc/common/libecalc/expression/expression_evaluator.py:360]: :D

File: src/ecalc/libraries/libecalc/common/tests/expression/test_expression_evaluator.py [hunk 3]
```diff
@@ -93,6 +93,12 @@ def test_lexer():
     token_values = [token.value for token in tokens]
     assert token_values == compare
 
+    expression = "2.4e6 {/} 100 {*} 1.35e-05"
```
Comment (TeeeJay) [inline src/ecalc/libraries/libecalc/common/tests/expression/test_expression_evaluator.py:96]: I would make a separate test. I see that we have fewer tests here than anticipated. Make sure you cover all edge cases. A good idea is to add parametric tests (my preference) to test both the case with scientific numbers alone, in an expression with other scientific numbers and combined with other tokentypes :)

Comment (markusrf) [reply src/ecalc/libraries/libecalc/common/tests/expression/test_expression_evaluator.py]: Have updated with more parametric tests now

**Review by TeeeJay — APPROVED** [review]
nice :)
